pyrobosim/pyrobosim/utils/pose.py

```diff
@@ -148,23 +178,26 @@ def from_dict(cls, pose_dict):
                     "roll": eul.get("roll", 0.0),
                 }
             )
+        args["angle_units"] = angle_units
         return cls(**args)
 
     @classmethod
-    def construct(self, data):
+    def construct(self, data, angle_units="radians"):
         """
         Constructs a pose object from any of the allowable input types.
 
         :param data: The input data describing the pose.
         :type data: dict[str, Any] or list or :class:`numpy.ndarray`
+        :param angle_units: Units for angle ('radians' or `degrees`). Default is 'radians'
+        :type angle_units: str
         :return: Pose object
         :rype: :class:`pyrobosim.utils.pose.Pose`
         raises ValueError: if the input data type is unsupported.
         """
         if isinstance(data, list):
-            return Pose.from_list(data)
+            return Pose.from_list(data, angle_units)
```

Comment:
To expand on my thinking: we could have the angle units used here, or remove it altogether